jpyinterpreter/src/main/java/org/optaplanner/jpyinterpreter/implementors/ExceptionImplementor.java

```diff
@@ -293,6 +303,54 @@ public static void startWith(int jumpTarget, FunctionMetadata functionMetadata,
         // cannot free, since try block store stack in locals => freeing enterResult messes up indexing of locals
     }
 
+    public static void startExceptBlock(FunctionMetadata functionMetadata, StackMetadata stackMetadata,
+            PythonExceptionTable.ExceptionBlock exceptionBlock) {
+        // In Python 3.11 and above, the stack here is
+        // [(stack-before-try), exception]
+
+        MethodVisitor methodVisitor = functionMetadata.methodVisitor;
+
+        // Stack is exception
+        // Duplicate exception to the current exception variable slot so we can reraise it if needed
+        stackMetadata.localVariableHelper.writeCurrentException(methodVisitor);
+        StackManipulationImplementor.restoreExceptionTableStack(functionMetadata, stackMetadata, exceptionBlock);
+
+        // Stack is (stack-before-try)
+        if (exceptionBlock.isPushLastIndex()) {
+            // Load 0 for last index since we don't use it
+            methodVisitor.visitFieldInsn(Opcodes.GETSTATIC, Type.getInternalName(PythonInteger.class), "ZERO",
+                    Type.getDescriptor(PythonInteger.class));
+        }
+
+        // Load exception
+        stackMetadata.localVariableHelper.readCurrentException(methodVisitor);
+        // Stack is (stack-before-try), index?, exception
+    }
+
+    public static void pushExcInfo(FunctionMetadata functionMetadata, StackMetadata stackMetadata) {
```

Comment:
Same comment as with `insn`. `Exception` is what the thing is called, `Exc` just brings inconsistency and new lingo that needs to be deciphered.

Reply:
This method name is delivered from the Python instruction name that it implements: `PUSH_EXC_INFO` (https://docs.python.org/3/library/dis.html#opcode-PUSH_EXC_INFO)